torch/_dynamo/variables/builtin.py

```diff
@@ -366,7 +366,20 @@ def call_function(
             proxy = tx.output.create_proxy(
                 "call_function", self.fn, *proxy_args_kwargs(args, kwargs)
             )
-            return DynamicShapeVariable.create(tx, proxy, None, **options)
+            value = None
+            if self.fn == range:
+                assert len(kwargs) == 0
+
+                def guard_if_dyn(arg):
+                    if isinstance(arg, DynamicShapeVariable):
+                        expr = arg.dyn_shape.get_pyobj().expr
+                        return tx.output.shape_env.evaluate_expr(expr)
```

Comment:
Does this actually guard 🤔

Reply:
it creates a useless guard like Eq(10, 10), I am speaking to Horace about why this is the way it is. Alternatively we could just extract an integer out without a guard.

Reply:
However, the value behind this dyn_shape might be an expression, and so guarding feels better than size_hint? maybe?

Reply:
To be clear, we better guard otherwise it's unsound.

Guard printing may be busted on branch as I hacked it to print symnodes as size hints